### DISCUSSION TITLE
NetAlias - RegEx Issue for Member parameter?

### DISCUSSION THREAD

**Discussion by jchinyou** [hnimminh/libresbc#81] [Q&A]
Hi All, 

Suspect this is something I'm doing wrong so figured I'd try and ask before raising an issue. 

When configuring the cluster it allows me to use '-' in the member names. However, when configuring the NetAlias side of the house my "member" parameter gets rejected with the '-'. 

Get Cluster: 
`
{
    "name": "itLabs",
    "members": [
        "itLabs-libresbc"
    ],
    "rtp_start_port": 10000,
    "rtp_end_port": 60000,
    "max_calls_per_second": 200,
    "max_concurrent_calls": 20
}
`

Post for Create NetAlias:
`
{
  "name": "Internet_Zone",
  "desc": "",
  "addresses": [
    {
      "member": "itLabs-libresbc",
      "listen": "10.0.xxx.xxx",
      "advertise": "x.xx.44.27"
    }
  ]
}
`

Response from Create NetAlias: 
`
{
    "detail": [
        {
            "loc": [
                "body",
                "addresses",
                0,
                "member"
            ],
            "msg": "string does not match regex \"^[a-zA-Z][a-zA-Z0-9_]+$\"",
            "type": "value_error.str.regex",
            "ctx": {
                "pattern": "^[a-zA-Z][a-zA-Z0-9_]+$"
            }
        }
    ]
}
`

Wondering if I am not allowed to use '-' should this have been rejected earlier or am I using an incorrect parameter for "member".

**Comment by hnimminh** [comment]
The node name was configured at installation step, and there no way to guarantee that `members` name will follow convention. 
I'm thinking to add the `guide note` for naming node name. may be need move naming node to API  instead of installation. (need time to consider)

best solution for now is rename your node follow pattern: "^[a-zA-Z][a-zA-Z0-9_]+$"

**Reply by hnimminh** [reply]
open #83

**Comment by ciscomonkey** [comment]
Was there any reason originally to not allow a dash in the ```member``` names?  I am actually hitting this myself.  My initial testing with modifying that regex appears to work, and I haven't found anything broken as a result yet.
